woocommerce/woocommerce
# Add Request Redirects tool that can re-route wp_get_remote calls

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This PR adds a new tool to the beta tester that can re-route outgoing requests via `wp_get_remote` function.
This tool is designed to test the remote inbox notification on staging.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

### Setup

1. Go to https://staging.woo.com admin (find the username and password from secret store 5926)
9. Go to Ads and publish a new inbox notification.
10. Confirm the added notification from https://staging.woocommerce.com/wp-json/wccom/inbox-notifications/2.0/notifications.json?locale=en_US

### Testing with Playground

1. [Create a Playground site](https://playground.wordpress.net/?blueprint-url=https://gist.githubusercontent.com/moon0326/ac60ad6389aedf86f8aeded3545faaaf/raw/7af79e663b7eae112c6cfb00ed7f1b8bb5d45301/gistfile1.txt)
2. Go to `Tools -> WCA Test Helper -> Request Redirects`
3. Click on `Edit`
4. Copy the password from secret store 5926 and update it.
5. Go to `Tools -> WCA Test Helper -> Tools` and run `Run wc_admin_daily job`
6. Go to `Tools -> WCA Test Helper -> Admin notes`
7. Click on `List all notes`
8. Confirm the note from the staging has been imported.


### Testing Locally


11. Checkout this branch locally.
12. Cd to `plugins/woocommerce-beta-tester` and run `pnpm run start` to build the changes.
13. Make sure to activate `woocommerce-beta-tester`
14. Open a new terminal session and use wp cli and run the following command.
```
wp option add wc-admin-test-helper-request-redirects '[{"original_endpoint":"https:\/\/woocommerce.com\/wp-json\/wccom\/inbox-notifications\/2.0\/notifications.json?locale=en_US","new_endpoint":"https:\/\/staging.woocommerce.com\/wp-json\/wccom\/inbox-notifications\/2.0\/notifications.json?locale=en_US","username":"woocommerce","password":"","enabled":true}]' --format=json
```
16. Go to `Tools -> WCA Test Helper -> Request Redirects`
17. Click on `Edit` button and update the password from secret store 5926
18. Go to `Tools -> WCA Test Helper -> Tools` and run `Run wc_admin_daily_job`
19. Go to `Tools -> WCA Test Helper -> Admin notes` and click on `List all notes` button.
20. Confirm the note from the staging has been imported.





### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

Add 'Request Redirects' tool to the beta tester to help re-routing wp_remote_get calls.

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
